### PR TITLE
[TechDocs] [Search] Rename/move common packages

### PR DESCRIPTION
**Pull request by iamEAP**
## Hey, I just made a Pull Request!

In order to align TechDocs/Search with [ADR011](https://backstage.io/docs/architecture-decisions/adrs-adr011), this PR...

- Renames/moves `@backstage/techdocs-common` -> `@backstage/plugin-techdocs-node`
- Renames/moves `@backstage/search-common` -> `@backstage/plugin-search-common`
- Maintains shell packages with the former names (re-exporting all) to help ease the transition

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/techdocs-empty-office.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/techdocs-common': patch
+'@backstage/plugin-techdocs-node': patch
```
Comment (Rugvip) [inline .changeset/techdocs-empty-office.md:3]: I think it could be nice to craft a separate changeset for the new packages tbh, without the deprecation label and so on :grin:

**Review by Rugvip — COMMENTED** [review]
Awesome, looks good, thanks! 👍 :grin:

File: .changeset/techdocs-empty-office.md (hunk 1 above)
Comment (iamEAP) [reply .changeset/techdocs-empty-office.md:3]: Gooood call.

**Comment by iamEAP** [comment]
One more 👀  and perhaps even a merge, when you have a chance @Rugvip?

File: packages/techdocs-common/package.json [hunk 2]
```diff
@@ -40,39 +40,9 @@
...
-  "devDependencies": {
-    "@backstage/cli": "^0.15.0",
-    "@types/fs-extra": "^9.0.5",
-    "@types/js-yaml": "^4.0.0",
-    "@types/mime-types": "^2.1.0",
-    "@types/mock-fs": "^4.13.0",
-    "@types/recursive-readdir": "^2.2.0",
-    "@types/supertest": "^2.0.8",
-    "supertest": "^6.1.3"
+    "@backstage/plugin-techdocs-node": "0.11.11"
```
Comment (Rugvip) [inline packages/techdocs-common/package.json:43]: You made the other one a `^` range, so I'm assuming you want one of those two

**Review by Rugvip — APPROVED** [review]
👌 , :shipit:

Found a minor optional thing but if needed fix before merge :grin:

File: packages/techdocs-common/package.json (hunk 2 above)
Comment (iamEAP) [reply packages/techdocs-common/package.json:43]: Ah, yeah.

Comment (iamEAP) [reply packages/techdocs-common/package.json:43]: Eagle eyes!
